### ISSUE TITLE
Data not shared in threaded Local Cluster

### ISSUE THREAD

**Issue by lanougue** [dask/distributed#2270]
Hi guys,
I am on a single machine with 28 cpus and would like to process a large array of data locally to avoid data transfert.
(This array comes from an other distributed computation over a much larger PBS Cluster).

Each part of the large array that can be processed independantly is stored (as a view) in args parameter and it run smoothly when I use delayed feature:
```
    myjobs=[]
    for arg in args:
        myjobs.append(dask.delayed(myfunc)(arg))
    with dask.config.set(scheduler='threads'):
        result = dask.compute(*myjobs)
```
meaning that no data seems to be copied/transfered to the threads. The threads seems to be able to read the large data array part without any big transfert.
However, I have no time gain comparatively to the serial code version with the non-parallelized map.

I concluded that only one thread was working and wasn't able to change the number of workers.
Adding num_workers=5 in the .compute() argument did not change anything.

I decided to use LocalCluster with threads to avoid data transfer :
```
locluster = LocalCluster(processes=False)
loclient = Client(locluster)
```
loclient returns : 1 worker & 56 cores
The code still run but huge amount of data seems to be transfered as suggested by the warning message :
>> UserWarning: Large object of size 1.16 GB detected in task graph: ...
and the process takes even more time.

I was thinking that one of the advantages of threads was that it is possible to avoid data transfert.
Specifing processes=False in LocalCluster initialization do not seems to work in the same way as the standard scheduler='threads' in dask.

Do you guys have some explanations of advices to run my code without data transfert ? I an hoping this is doable since I am on a single machine.

Thanks

**Comment by mrocklin** [comment]
> However, I have no time gain comparatively to the serial code version with the non-parallelized map

There are many reasons why your computation would not be accelerated with threads:

1.  Maybe it is bound by some other piece of hardware, like disk speeds, network bandwidth, or memory hierarchy
2.  Maybe your computation is bound by the GIL

**Comment by lanougue** [comment]
Thanks mrocklin,
I looked into this GIL stuff which effectively looks like a big limitation for python multi-threading.
I only use python code in the called function which probably not release that famous GIL.
I will try to release it using more sophisticated numpy manipulations.

Thanks again

**Comment by mrocklin** [comment]
Yes, using Numpy would probably help quite a bit here.  Possibly enough that you won't need Dask.
